decoder/main.go

```diff
@@ -399,33 +402,33 @@ func UpdatePokemonBatch(ctx context.Context, db db.DbDetails, scanParameters Sca
 		if err != nil {
 			log.Printf("getOrCreatePokemonRecord: %s", err)
 		} else {
-			pokemon.updateFromMap(ctx, db, mapPokemon.Data, int64(mapPokemon.Cell), username)
+			pokemon.updateFromMap(ctx, db, mapPokemon.Data, int64(mapPokemon.Cell), weatherLookup, mapPokemon.Timestamp, username)
 			storedDiskEncounter := diskEncounterCache.Get(encounterId)
 			if storedDiskEncounter != nil {
 				diskEncounter := storedDiskEncounter.Value()
 				diskEncounterCache.Delete(encounterId)
 				pokemon.updatePokemonFromDiskEncounterProto(ctx, db, diskEncounter, username)
-				log.Infof("Processed stored disk encounter")
+				//log.Infof("Processed stored disk encounter")
 			}
-			savePokemonRecord(ctx, db, pokemon)
+			savePokemonRecordAsAtTime(ctx, db, pokemon, false, mapPokemon.Timestamp/1000)
 		}
 		pokemonMutex.Unlock()
 	}
 }
 
-func UpdateClientWeatherBatch(ctx context.Context, db db.DbDetails, p []RawClientWeatherData) {
```

Comment:
Why did you remove the RawClientWeatherData?! not necessary anymore? normally we collect all proto we need and pass the struct array :)

Reply:
Everything is in the proto so no need to create an additional struct. Unnecessary.